src/background/services/openPayments.ts

```diff
@@ -424,31 +424,49 @@ export class OpenPaymentsService {
     if (calculatedHash !== hash) throw new Error('Invalid interaction hash')
   }
 
-  private async confirmPayment(url: string): Promise<InteractionParams> {
+  private async closeTab(currentTab: number, tabToClose: number) {
+    await this.browser.tabs.update(currentTab, { active: true })
+    await this.browser.tabs.remove(tabToClose)
+  }
+
+  private async getInteractionInfo(url: string): Promise<InteractionParams> {
     const currentTab = await getCurrentActiveTab(this.browser)
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this
```

Comment:
Not sure if this is the best approach. Since we are now using a named function for the callback - to remove the listener after the interaction is complete, we cannot reference the correct `this`.

Reply:
We can `const getInteractionRefAndHash = async (...)` and then use it within with `this.browser.tabs.onUpdated.removeListener(getInteractionRefAndHash)` without the `self` assignment (it won't cause use before define issues). Use named function isn't necessary.

Reply:
```ts
type Callback = Parameters<Tabs.onUpdatedEvent["addListener"]>[0]
const getInteractionRefAndHash: Callback = async (tabId: number, changeInfo) => {
  if (tabId !== tab.id) return;
  // ....
  await this.closeTab(currentTab.id!, tabId);
  this.browser.tabs.onUpdated.removeListener(getInteractionRefAndHash);
};
// ...

this.browser.tabs.onUpdated.addListener(getInteractionRefAndHash);

```